mantidproject/mantid
# if ws is rebinnable, use it, else use its original ws

**Description of work.**
PR addresses [this defect](https://code.ornl.gov/sns-hfir-scse/diffraction/single-crystal/single-crystal-diffraction/-/issues/275)

- [x] `Framework/PythonInterface/plugins/algorithms/ConvertQtoHKLMDHisto.py ` modified
- [x] pr against `ornl-next`
- [x] pr against `release-next` 

Companion to PR's [ornl-next [![GitHub issue/pull request detail](https://img.shields.io/github/issues/detail/state/mantidproject/mantid/31333)](https://github.com/mantidproject/mantid/pull/31333) , release-next [![GitHub issue/pull request detail](https://img.shields.io/github/issues/detail/state/mantidproject/mantid/31360)](https://github.com/mantidproject/mantid/pull/31360)]




**To test:**

Launch Mantid workbench and run the following script from the UI:
```# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

CreateSimulationWorkspace(Instrument='TOPAZ',
                          BinParams='1,100,10000',
                          UnitX='TOF',
                          OutputWorkspace='cws')
                          
omega, chi, phi = 20, 40, 30

AddSampleLog('cws', "omega", str(omega), "Number")
AddSampleLog('cws', "chi", str(chi), "Number")
AddSampleLog('cws', "phi", str(phi), "Number")

SetGoniometer('cws', Axis0='omega,0,1,0,1', Axis1='chi,0,0,1,1', Axis2='phi,0,1,0,1')

md = ConvertToMD(InputWorkspace='cws', 
                 QDimensions='Q3D',
                 dEAnalysisMode='Elastic', 
                 Q3DFrames='Q_sample',
                 QConversionScales='Q in A^-1',
                 LorentzCorrection='1', 
                 MinValues='-5,-5,-5', 
                 MaxValues='5,5,5')
                 
FakeMDEventData(md, UniformParams='1000')

SetUB(md, u='1,1,0', v='0,0,1', a=5, b=5, c=6, alpha=90, beta=90, gamma=120)

HKL = ConvertQtoHKLMDHisto(md)
HKL *= 2
```
Then right click and open the generated HKL workspace in the slice viewer.  Attempt to zoom and observe no error is thrown as well as it functioning correctly.

#### Release Notes ####

Diffraction Release Notes 6.1
* fixed changes in ConvertQtoHKLMDHisto to accomodate zoom in slice viewer, as a result this algorithm will now drop the connection to the original workspace

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
